cpp/tests/utilities_tests/multibuffer_memset_tests.cpp

```diff
@@ -0,0 +1,115 @@
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/cudf_gtest.hpp>
+#include <cudf_test/type_lists.hpp>
+#include <src/io/utilities/multibuffer_memset.hpp>
+#include <thrust/iterator/transform_iterator.h>
+#include <type_traits>
+#include <cudf/io/parquet.hpp>
+
+template <typename T>
+struct MultiBufferTestIntegral : public cudf::test::BaseFixture {};
+
+TEST(MultiBufferTestIntegral, BasicTest)
+{
+    long NUM_BUFS = 1;
+    std::vector<long> BUF_SIZES{600};
```

Comment:
For all these tests, I think a better strategy would be:
- Create the buffer on the host.
- Add 1k of padding before and after each actual buffer to be memset.
- Manually memset the pad space to `0xff` on the host.
- Manually memset the non-pad space to `0xee`
- Copy to the gpu
- Do the zeroing memset
- Copy back to the host, verify that that pad space is untouched and the non-pad space is properly set to 0.